src/main/java/com/minecolonies/util/BlockUtils.java

```diff
@@ -106,4 +107,140 @@ public static boolean isWater(IBlockState iBlockState)
         return Objects.equals(iBlockState, Blocks.water.getDefaultState())
                || Objects.equals(iBlockState, Blocks.flowing_water.getDefaultState());
     }
+
+    /**
+     * Gets an item from a block.
+     * @param block input block.
+     * @return output item.
+     */
+    public static Item getItemFromBlock(Block block)
+    {
+        Item item = null;
+
+        if(block instanceof BlockBanner)
+        {
+            item = Items.banner;
+        }
+        else if(block instanceof BlockBed)
+        {
+            item = Items.bed;
+        }
+        else if(block instanceof BlockBrewingStand)
+        {
+            item = Items.brewing_stand;
+        }
+        else if(block instanceof BlockCake)
+        {
+            item = Items.cake;
+        }
+        else if(block instanceof BlockCauldron)
+        {
+            item = Items.cauldron;
+        }
+        else if(block instanceof BlockCocoa)
+        {
+            item = Items.dye;
+        }
+        else if(block instanceof BlockCrops)
+        {
+            if(block instanceof BlockCarrot)
+            {
+                item = Items.carrot;
+            }
+            else if(block instanceof BlockPotato)
+            {
+                item = Items.potato;
+            }
+            else
+            {
+                item = Items.wheat_seeds;
+            }
+        }
+        else if(block instanceof BlockDaylightDetector)
+        {
+            item = Item.getItemFromBlock(Blocks.daylight_detector);
+        }
+        else if(block instanceof BlockDoor)
+        {
+            item = block == Blocks.iron_door ? Items.iron_door
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (6) used in the expression (maximum allowed 4). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1067)